sklearn/metrics/_plot/cap_curve.py

```diff
@@ -0,0 +1,274 @@
+
+        response_method : {'predict_proba', 'decision_function', 'auto'} \
+                default='auto'
+            Specifies whether to use :term:`predict_proba` or
+            :term:`decision_function` as the target response. If set to 'auto',
+            :term:`predict_proba` is tried first and if it does not exist
+            :term:`decision_function` is tried next.
+
+        pos_label : int, float, bool or str, default=None
+            The class considered as the positive class when computing metrics.
+            By default, `estimators.classes_[1]` is considered as the positive class.
+
+        name : str, default=None
+            Name of CAP Curve for labeling. If `None`, use the name of the estimator.
+
+        ax : matplotlib axes, default=None
+            Axes object to plot on. If `None`, a new figure and axes is created.
+
+        **kwargs : dict
+            Keyword arguments to be passed to matplotlib's `plot`.
+
+        Returns
+        -------
+        display : :class:`~sklearn.metrics.CumulativeAccuracyDisplay`
+            The CAP Curve display.
+        """
+        
+        # validate and prepare the prediction scores
+        if response_method == "auto":
+            if hasattr(estimator, "predict_proba"):
+                response_method = "predict_proba"
+            elif hasattr(estimator, "decision_function"):
+                response_method = "decision_function"
+            else:
+                raise ValueError("Estimator does not have a predict_proba or decision_function method.")
+
+        if response_method == "predict_proba":
+            probabilities = estimator.predict_proba(X)
+            
+            # assuming positive class is the second column
+            if pos_label is None:
+                pos_label = 1
+            class_index = np.where(estimator.classes_ == pos_label)[0][0]
+            y_pred = probabilities[:, class_index]
+        elif response_method == "decision_function":
+            y_pred = estimator.decision_function(X)
+
+        if name is None:
+            name = estimator.__class__.__name__
+
+        return cls.from_predictions(
+            y_true=y,
+            y_pred=y_pred,
+            sample_weight=sample_weight,
+            name=name,
+            ax=ax,
+            pos_label=pos_label,
+            **kwargs,
+        )
+
+
+
+import matplotlib.pyplot as plt
+from sklearn.datasets import make_classification
+from sklearn.model_selection import train_test_split
+from sklearn.linear_model import LogisticRegression
+
+X, y = make_classification(n_samples=1000, n_features=20, n_classes=2, random_state=42)
+X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)
+clf = LogisticRegression(max_iter=1000)
+clf.fit(X_train, y_train)
+y_scores = clf.decision_function(X_test)
+
+CumulativeAccuracyDisplay.from_predictions(y_test, y_scores, name='Logistic Regression')
+plt.title('CAP Curve using from_predictions')
+plt.show()
```

Comment:
For information when running this code I get:

<img width="622" alt="image" src="https://github.com/scikit-learn/scikit-learn/assets/89061/bd2102cc-c350-425f-9b68-db5ead3dde4d">


This looks good but I would like to also have an option to display the curve for oracle predictions and the curve for the random predictions baseline.

Reply:
In addition, I think we should set better the x- and y-lim (as in ROC curve display).
Also, is it common for this curve to be be in a squared axis? I see this is kind of different from the ROC and PR curve in this regards, where the x- and y-axis are not the same unit.

Reply:
> An option for relative VS absolute scales sounds relevant to me. Don't have a opinion of which should be the default.

I would use relative intuitively as the default.